shared/core/types/relay_test.go

```diff
@@ -0,0 +1,137 @@
+package types
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestRelayValidate(t *testing.T) {
+	testCases := []struct {
+		name     string
+		relay    Relay
+		expected error
+	}{
+		{
+			name: "valid Relay: JSONRPC",
+			// IMPROVE: use a factory function to build test relays
```

Comment:
I think this `IMPROVE` belongs in `relay.go` instead of test, no?

Reply:
Yes it does. Thank you. Fixed.